br/pkg/lightning/errormanager/errormanager_test.go

```diff
@@ -261,28 +297,201 @@ func TestReplaceConflictKeysIndexKvChecking(t *testing.T) {
+		WillReturnRows(sqlmock.NewRows([]string{"raw_key", "index_name", "raw_value", "raw_handle"}).
+			AddRow(data1IndexKey, "uni_b", data1IndexValue, data1RowKey).
+			AddRow(data1IndexKey, "uni_b", data2IndexValue, data2RowKey))
+	mockDB.ExpectQuery("\\QSELECT raw_key, raw_value, raw_handle FROM `lightning_task_info`.conflict_error_v1 WHERE table_name = ? AND index_name = 'PRIMARY' ORDER BY raw_key\\E").
+		WillReturnRows(sqlmock.NewRows([]string{"raw_key", "raw_value", "raw_handle"}))
+
+	cfg := config.NewConfig()
+	cfg.TikvImporter.DuplicateResolution = config.DupeResAlgReplace
+	cfg.App.TaskInfoSchemaName = "lightning_task_info"
+	em := New(db, cfg, log.L())
+	err = em.Init(ctx)
+	require.NoError(t, err)
+
+	fnGetLatestCount := atomic.NewInt32(0)
+	fnDeleteKeyCount := atomic.NewInt32(0)
+	pool := utils.NewWorkerPool(16, "resolve duplicate rows by replace")
+	err = em.ReplaceConflictKeys(
+		ctx, tbl, "test", pool,
+		func(ctx context.Context, key []byte) ([]byte, error) {
+			fnGetLatestCount.Add(1)
+			switch {
+			case bytes.Equal(key, data1IndexKey):
+				return data1IndexValue, nil
+			case bytes.Equal(key, data2RowKey):
+				return data2RowValue, nil
+			default:
+				return nil, fmt.Errorf("key %v is not expected", key)
+			}
+		},
+		func(ctx context.Context, key []byte) error {
+			fnDeleteKeyCount.Add(1)
+			if !bytes.Equal(key, data2RowKey) {
+				return fmt.Errorf("key %v is not expected", key)
+			}
+			return nil
+		},
+	)
+	require.NoError(t, err)
+	require.Equal(t, int32(3), fnGetLatestCount.Load())
+	require.Equal(t, int32(1), fnDeleteKeyCount.Load())
+	err = mockDB.ExpectationsWereMet()
+	require.NoError(t, err)
+}
+
+func TestReplaceConflictKeys(t *testing.T) {
+	column1 := &model.ColumnInfo{
+		ID:           1,
+		Name:         model.NewCIStr("a"),
+		Offset:       0,
+		DefaultValue: 0,
+		FieldType:    *types.NewFieldType(mysql.TypeLong),
+		Hidden:       true,
+		State:        model.StatePublic,
+	}
+	column1.AddFlag(mysql.PriKeyFlag)
+
+	column2 := &model.ColumnInfo{
+		ID:           2,
+		Name:         model.NewCIStr("b"),
+		Offset:       1,
+		DefaultValue: 0,
+		FieldType:    *types.NewFieldType(mysql.TypeLong),
+		Hidden:       true,
+		State:        model.StatePublic,
+	}
+	column2.AddFlag(mysql.UniqueKeyFlag)
+
+	column3 := &model.ColumnInfo{
+		ID:           3,
+		Name:         model.NewCIStr("c"),
+		Offset:       2,
+		DefaultValue: 0,
+		FieldType:    *types.NewFieldType(mysql.TypeBlob),
+		Hidden:       true,
+		State:        model.StatePublic,
+	}
+
+	index := &model.IndexInfo{
+		ID:    1,
+		Name:  model.NewCIStr("uni_b"),
+		Table: model.NewCIStr(""),
+		Columns: []*model.IndexColumn{
+			{
+				Name:   model.NewCIStr("b"),
+				Offset: 1,
+				Length: -1,
+			}},
+		Unique:  true,
+		Primary: false,
+		State:   model.StatePublic,
+	}
+
+	table := &model.TableInfo{
```

Comment:
can use this to mock table info from sql

https://github.com/pingcap/tidb/blob/2680a9a8ea148c5dfefe046512b587bc8880816d/disttask/importinto/encode_and_sort_operator_test.go#L190C1-L194C71

Reply:
we need to use `ddl.MockTableInfo` for this method, but if we import `github.com/pingcap/tidb/ddl`, it would result in "import cycle not allowed in test"

```
# github.com/pingcap/tidb/br/pkg/lightning/errormanager
FAIL	github.com/pingcap/tidb/br/pkg/lightning/errormanager [setup failed]
package github.com/pingcap/tidb/br/pkg/lightning/errormanager
	imports github.com/pingcap/tidb/ddl
	imports github.com/pingcap/tidb/ddl/ingest
	imports github.com/pingcap/tidb/br/pkg/lightning/errormanager: import cycle not allowed in test
FAIL
````